src/nomad_material_processing/general.py

```diff
@@ -444,41 +448,215 @@ class IrregularParallelSurfaces(Geometry):
+        This corresponds to taking the coordinates of the first lattice node on that
+        direction after the origin as direction indices.
+        When a primitive unit cell is used, the direction indices are all integer;
+        they may instead be rational when a centred unit cell is adopted.
+        Ref. https://dictionary.iucr.org/Direction_indices""",
+    )
+
+
+class MiscutOrientation(ArchiveSection):
+    """
+    This direction can be described by a crystallographic direction
+    [hkl], which indicates the direction of the tilt relative to the crystal axes.
+
+    The miscut might be directed in a non-pure crystallographic direction.
+    In this case two components must be specified,
+    either in Cartesian or polar coordinates.
+    """
+
+
+class CartesianMiscutOrientation(MiscutOrientation):
+    """
+    This direction can be described by a crystallographic direction
+    [hkl], which indicates the direction of the tilt relative to the crystal axes.
+
+    The miscut might be directed in a non-pure crystallographic direction.
+    In this case two components must be specified,
+    either in Cartesian or polar coordinates.
+    """
+
+    orientation = SubSection(
+        section_def=CrystallographicDirection,
+    )
+    orientation_perp = SubSection(
+        section_def=CrystallographicDirection,
+        description='A direction perpendicular to the first given',
+    )
+
+
+class PolarMiscutOrientation(MiscutOrientation):
+    """
+    This direction can be described by a crystallographic direction
+    [hkl], which indicates the direction of the tilt relative to the crystal axes.
+
+    The miscut might be directed in a non-pure crystallographic direction.
+    In this case two components must be specified,
+    either in Cartesian or polar coordinates.
+    """
+
+    orientation = SubSection(
+        section_def=CrystallographicDirection,
+    )
+    rho = Quantity(
+        type=float,
+        unit='meter',
+        description="""
+        Module of polar coordinates
+        """,
+        a_eln=ELNAnnotation(
+            component=ELNComponentEnum.NumberEditQuantity,
+            defaultDisplayUnit='nanometer',
+        ),
+    )
+
+    theta = Quantity(
+        type=float,
+        unit='degree',
+        description="""
+        Angle of the miscut direction relative to the main orientation
+        given in this section.
+        """,
+        a_eln=ELNAnnotation(
+            component=ELNComponentEnum.NumberEditQuantity,
+        ),
+    )
+
+
 class Miscut(ArchiveSection):
     """
     The miscut in a crystalline substrate refers to
     the intentional deviation from a specific crystallographic orientation,
     commonly expressed as the angular displacement of a crystal plane.
+
+    It can either be specified as directed along one of the crystallographic axes or
+    in a specific direction in the crystal defined by a main direction and the
+    perpendicular one.
     """
 
     angle = Quantity(
```

Comment:
Hrmm, now I see that you left that out here. I think I'm a bit confused as to where you want to store the magnitude of the miscut and where you want to store the direction of the miscut.